File: 1813-maximum-erasure-value/maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        int left = 0;
        int maxSum = 0;
        int currSum = 0;

        for (int right = 0; right < nums.length; right++){
            while (seen.contains(nums[right])){
                seen.remove(nums[left]);
                currSum -= nums[left];
                left++;
            }
            seen.add(nums[right]);
            currSum += nums[right];
            maxSum = Math.max(maxSum, currSum);

        }
        return maxSum;
    }
}

File: 1813-maximum-erasure-value/maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        left = 0 
        maxSum = 0
        currSum = 0
        seen = set()

        for right in range(len(nums)):
            while nums[right] in seen:
                currSum -= nums[left]
                seen.remove(nums[left])
                left += 1

            seen.add(nums[right])
            currSum += nums[right]
            maxSum = max(maxSum, currSum)

        return maxSum 